src/OrleansSQLUtils/Storage/RelationalStorage.cs

```diff
@@ -195,35 +201,55 @@ private RelationalStorage(string invariantName, string connectionString)
         }
 
 
-        private static async Task<Tuple<IEnumerable<TResult>, int>> ExecuteReaderAsync<TResult>(DbCommand command, Func<IDataRecord, int, TResult> selector)
+        private async Task<Tuple<IEnumerable<TResult>, int>> ExecuteReaderAsync<TResult>(DbCommand command, Func<IDataRecord, int, CancellationToken, Task<TResult>> selector, CancellationToken cancellationToken, CommandBehavior commandBehavior)
         {
-            using(var reader = await command.ExecuteReaderAsync().ConfigureAwait(continueOnCapturedContext: false))
-            {
-                return await SelectAsync(reader, selector).ConfigureAwait(false);
+            using(var reader = await command.ExecuteReaderAsync(commandBehavior, cancellationToken).ConfigureAwait(continueOnCapturedContext: false))
+            {                                                                
```

Comment:
I added offloading with Task.Run if the underluing ADO.NET provider is a synchronous one. Currently only MySQL is flagged as syncrnous, but likely there are others too (Oracle?).